apps/condo/domains/miniapp/schema/preprocessors/B2BAppAccess.js

```diff
@@ -0,0 +1,505 @@
+        if (typeof defaultAccess.delete !== 'undefined') newAccess.delete = wrapInCustomAccess(defaultAccess.delete, customAccessFn)
+        if (typeof defaultAccess.auth !== 'undefined') newAccess.auth = wrapInCustomAccess(defaultAccess.auth, customAccessFn)
+        return newAccess
+    }
+
+    throw new Error(
+        `fieldAccessWrapperIfNeeded(), received ${type}.`,
+    )
+}
+
+/**
+ *
+ * @param pathToOrganizationId {string[]}
+ * @return {string}
+ *
+ * @example
+ * Input: ['a', 'b', 'c']
+ * Output: '{ a { b { c } } }'
+ */
+const generateGqlDataPart = (pathToOrganizationId) => {
+    if (!pathToOrganizationId || !isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) throw new Error('"pathToOrganizationId" should not be empty array')
+    return `{ ${pathToOrganizationId.join(' { ')}${' }'.repeat(pathToOrganizationId.length)}`
+}
+
+/**
+ *
+ * @param listKey
+ * @param pathToOrganizationId
+ * @return {string}
+ *
+ * @example
+ * Input: ('Ticket', ['organization', 'id'])
+ * Output: `
+ *         query getAllTickets ($where: TicketWhereInput, $first: Int = 1) {
+ *             objs: allTickets(where: $where, first: $first) { organization { id } }
+ *         }
+ *     `
+ */
+const generateGqlQueryAsString = (listKey, pathToOrganizationId) => {
+    if (!isString(listKey) || listKey.trim().length < 1) throw new Error('"listKey" must not be empty string!')
+    if (!pathToOrganizationId || !isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) throw new Error('"pathToOrganizationId" should not be empty array')
+
+    return `
+        query getAll${pluralize.plural(listKey)} ($where: ${listKey}WhereInput, $first: Int = 1) {
+            objs: all${pluralize.plural(listKey)}(where: $where, first: $first) ${generateGqlDataPart(pathToOrganizationId)}
+        }
+    `
+}
+
+const ALL_GENERATED_GQL_QUERIES = new Map()
+
+/**
+ *
+ * @param pathToOrganizationId {string[]}
+ * @return {*|DocumentNode}
+ *
+ * @example
+ * Input: ['ticket', 'organization', 'id']
+ * Output: gql`
+ *         query getAllTickets ($where: TicketWhereInput, $first: Int = 1) {
+ *             objs: allTickets(where: $where, first: $first) { organization { id } }
+ *         }
+ *     `
+ */
+const generateGqlQuery = (pathToOrganizationId) => {
+    if (!pathToOrganizationId || !isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) throw new Error('"pathToOrganizationId" should not be empty array')
+    if (pathToOrganizationId.length < 2) throw new Error(`To generate gql "pathToOrganizationId" must contain at least two elements! But was ${pathToOrganizationId}`)
+
+    const gqlKey = pathToOrganizationId.join(':')
+
+    // get already generated gql query to reuse them
+    if (ALL_GENERATED_GQL_QUERIES.has(gqlKey)) {
+        return ALL_GENERATED_GQL_QUERIES.get(gqlKey)
+    }
+
+    const listKey = pathToOrganizationId[0].charAt(0).toUpperCase() + pathToOrganizationId[0].slice(1)
```

Comment:
field name may not match with listkey

Reply:
Now this is taken automatically from schema field